bin/generate_package_icons.dart

```diff
@@ -9,50 +9,98 @@ import 'package:dart_style/dart_style.dart';
+    ]));
+
+  final emitter = DartEmitter();
+  final generatedFileContent = DartFormatter().format(
+    '${phosphorLib.accept(emitter)}',
+  );
+
+  saveContentToFile(
+    filePath: '../lib/src/phosphor_icons.dart',
+    content: generatedFileContent,
+  );
+}
+
+/// Generates an abstract class that exposes all the icons that every style extends
+void generateBaseClass(List icons) {
+  final styles = StyleFileData.values;
+  final stylesEnum = Enum(
+    (enumBuilder) => enumBuilder
+      ..name = 'PhosphorIconsStyle'
+      ..values.addAll(
         styles.map(
-          (style) {
-            return Field(
-              (fieldBuilder) => fieldBuilder
-                ..docs.add(style.docsLine)
-                ..static = true
-                ..modifier = FieldModifier.final$
-                ..name = style.styleName
-                ..assignment = Code(style.classConstructorLine),
-            );
-          },
-        ).toList()
-          ..sort((a, b) => a.name.compareTo(b.name)),
+          (style) => EnumValue((builder) => builder
+            ..name = style.styleName
+            ..docs.add(style.docsLine)),
+        ),
       ),
   );
+  final methods = icons.map((icon) => buildBaseFieldIcon(icon)).toList();
+  final phosphorIconsClass = Class((classBuilder) {
+    classBuilder
+      ..name = 'PhosphorIcons'
+      ..methods.addAll(methods);
+  });
 
   final phosphorLib = Library(
     (libraryBuilder) => libraryBuilder
-      ..directives.addAll(
-        styles.map(
+      ..directives.addAll([
+        Directive.import(
+          'package:phosphor_flutter/src/phosphor_icon_data.dart',
+        ),
+        ...styles.map(
           (style) => Directive.import(
             'package:phosphor_flutter/src/${style.classFileName}',
           ),
         ),
-      )
-      ..body.add(phosphorIconsClass),
+      ])
+      ..body.addAll([phosphorIconsClass, stylesEnum]),
   );
 
   final emitter = DartEmitter();
   final generatedFileContent = DartFormatter().format(
     '${phosphorLib.accept(emitter)}',
   );
-
   saveContentToFile(
-    filePath: '../lib/src/phosphor_icons.dart',
+    filePath: '../lib/src/phosphor_icons_base.dart',
     content: generatedFileContent,
   );
 }
 
+Method buildBaseFieldIcon(dynamic icon) {
+  final properties = icon['properties'] as Map<String, dynamic>;
+  final fullName = properties['name'] as String;
+  final name = formatName(fullName, style: 'regular');
+  final styles = StyleFileData.values;
+  var code = 'switch (style) {';
+  for (final style in styles) {
+    code += '''
+case PhosphorIconsStyle.${style.styleName}:
+  return ${style.className}.$name;
+''';
+  }
+  code += '}';
+  return Method((methodBuilder) => methodBuilder
+    ..name = name
+    ..static = true
+    ..requiredParameters.add(Parameter((parameterBuilder) => parameterBuilder
+      ..name = 'style'
+      ..type = Reference('PhosphorIconsStyle')))
+    ..body = Code(code)
+    ..returns = Reference('PhosphorIconData'));
```

Comment:
add docs to the methods, preferable with a preview of the icon, could be a preview only of the regular one, but if you could add all that would be great like the [`BoxFit` enum](https://api.flutter.dev/flutter/painting/BoxFit.html)